Library/Homebrew/completions.rb

```diff
@@ -296,7 +296,15 @@ def generate_fish_subcommand_completion(command)
           next unless FISH_NAMED_ARGS_COMPLETION_FUNCTION_MAPPING.key? type
 
           named_arg_function = FISH_NAMED_ARGS_COMPLETION_FUNCTION_MAPPING[type]
-          named_args << "__fish_brew_complete_arg '#{command}' -a '(#{named_arg_function})'"
+          named_arg_prefix = "complete -f -c brew -n '__fish_brew_command #{command}; and not __fish_seen_argument"
+
+          named_args << if type == :formula && named_args_types.include?(:cask)
+            "#{named_arg_prefix} -l cask -l casks' -a '(#{named_arg_function})'"
```

Comment:
```suggestion
            "__fish_brew_complete_arg '#{command}; and not __fish_seen_argument -l cask -l casks' -a '(#{named_arg_function})
```

Will this work? It looks like the [`__fish_brew_complete_arg `](https://github.com/Homebrew/brew/blob/8b677c0ddd5f96b5521c1b353a9ca07dfe57b80e/Library/Homebrew/completions/fish.erb#L188-L192) function can accept any condition. This seems to work locally for me.

Reply:
`complete` only takes a single `-n` argument, and if multiple ones are provided, the last one will be used. So if using `__fish_brew_complete_arg`, the formulae names will overflow every `brew` completions 😅

Reply:
Maybe I'm misunderstanding, but where is the second `-n`?

I'm picturing a line that looks like:

```fish
__fish_brew_complete_arg 'install; and not __fish_seen_argument -l cask -l casks' -a '(__fish_brew_suggest_formulae_all)'
```

Which I think should "simplify to":

```fish
complete -f -c brew -n "__fish_brew_command install; and not __fish_seen_argument -l cask -l casks" -a '(__fish_brew_suggest_formulae_all)'
```

Isn't that identical to what's currently in this PR?

Reply:
@kidonng just following up on this. If you don't have time I'll probably finish this one up myself

Reply:
😅 Sorry (again) for the wait! I will follow up in one or two days.